### DISCUSSION TITLE
Dice Roller in Statblocks

### DISCUSSION THREAD

**Discussion by XintakXVI** [ebullient/ttrpg-convert-cli#121] [Q&A] [answered]
Is there any way to get dice roller functionality in the monster statblocks?
Thanks!

**Comment by ebullient** [comment]
I've thought about adding an option for this, that said are you talking about YAML-based statblocks or the formatted statblock admonitions?

**Reply by XintakXVI** [reply]
The admonitions one. I followed the [instructions](https://github.com/ebullient/ttrpg-convert-cli#recommended-plugins:~:text=Create%20a%20custom%20admonition%20called%20statblock%20(I%20recommend%20a%20dragon%20icon)) under the README.

Just for my knowledge, YAML-based statblocks would refer to those from the TTRPG Statblocks plugin right?

**Reply by ebullient** [reply]
yep. I've been wondering when someone would ask for dice roller in the admonition variant. I find them noisy.. but I'll add the option. ;)

**Reply by XintakXVI** [reply]
Wow, thanks!
Looking forward to that functionality!
I hope it's not too difficult to integrate 😅

**Reply by ebullient** [reply, marked as answer]
Ok. Two ways (as of [1.1.1](https://github.com/ebullient/ttrpg-convert-cli/tree/1.2.0#111-dice-roller-in-statblocks-and-text)):

1. Set `useDiceRoller` in your config options and use the long-form / template-based statblock, like [this (default)](https://github.com/ebullient/ttrpg-convert-cli/blob/1.2.0/src/main/resources/templates/tools5e/monster2md.txt) or [this](https://github.com/ebullient/ttrpg-convert-cli/blob/main/examples/templates/tools5e/images-monster2md.txt).
2. Use the Fantasy statblock plugin to render statblocks, and use a [template](https://github.com/ebullient/ttrpg-convert-cli/blob/main/examples/templates/tools5e/monster2md-yamlStatblock-body.txt) that will include the yaml that the statblock plugin needs.
